core/src/main/java/org/apache/iceberg/rest/responses/BaseScanTaskResponse.java

```diff
@@ -23,13 +23,16 @@
 import org.apache.iceberg.DeleteFile;
 import org.apache.iceberg.FileScanTask;
 import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 import org.apache.iceberg.rest.RESTResponse;
+import org.apache.iceberg.util.DeleteFileSet;
 
 public abstract class BaseScanTaskResponse implements RESTResponse {
 
   private final List<String> planTasks;
   private final List<FileScanTask> fileScanTasks;
-  private final List<DeleteFile> deleteFiles;
+  private final DeleteFileSet deleteFiles;
```

Comment:
I also think we should be able to get rid of this local state in the response as well once the deprecated API is removed. The only reason it exists, is just for the getter to pass back into implementations of BaseScanTaskResponse. We should also deprecate the public getter for deleteFiles imo.

Reply:
Once we remove the getter, we can get rid of the constructors passing in the deleteFiles too, luckily those are package private